.github/workflows/on_build_do_publish_image.yml

```diff
@@ -0,0 +1,82 @@
+name: On Build Do Publish Image
+on:
+   workflow_run:
+      workflows: [On PR Open or Push Do Build]
+      types: [completed]
+env:
+   IMAGE_NAME: ${{ vars.QUAY_IMAGE_NAME  || 'quay.io/infinispan-test/server' }}
+
+concurrency:
+  # Only cancel jobs for PR updates
+  group: ci-publish-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  get-info:
+    name: "Get information about the source run"
+    runs-on: ubuntu-latest
+    outputs:
+      sourceHeadRepo: ${{ steps.workflow-run-info.outputs.sourceHeadRepo }}
+      sourceHeadBranch: ${{ steps.workflow-run-info.outputs.sourceHeadBranch }}
+      sourceHeadSha: ${{ steps.workflow-run-info.outputs.sourceHeadSha }}
+      mergeCommitSha: ${{ steps.workflow-run-info.outputs.mergeCommitSha }}
+      targetCommitSha: ${{ steps.workflow-run-info.outputs.targetCommitSha }}
+      pullRequestNumber: ${{ steps.workflow-run-info.outputs.pullRequestNumber }}
+      pullRequestLabels: ${{ steps.workflow-run-info.outputs.pullRequestLabels }}
+      targetBranch: ${{ steps.workflow-run-info.outputs.targetBranch }}
+      sourceEvent: ${{ steps.workflow-run-info.outputs.sourceEvent }}
+    steps:
+      - name: "Get information about the current run"
+        uses: potiuk/get-workflow-origin@v1_1
+        id: workflow-run-info
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          sourceRunId: ${{ github.event.workflow_run.id }}
+
+  get-server-version:
+   name: "Get server version"
+   runs-on: ubuntu-latest
+   outputs:
+      imageName: ${{ steps.step1.outputs.imageName }}
+   steps:
+      - name: Download Server version
+        uses: actions/download-artifact@v4
+        with:
+         name: server-version
+         run-id: ${{ github.event.workflow_run.id }}
+         github-token: ${{ github.token }}
+      - name: Propagate env vars
+        id: step1
+        run: echo "imageName=$IMAGE_NAME" >> $GITHUB_OUTPUT
+
+  images-branch:
+      if: needs.get-info.outputs.sourceEvent == 'push'
+      needs:
+         - get-info
+         - get-server-version
+      uses: infinispan/infinispan-images/.github/workflows/publish_image.yml@main
+      secrets:
+         token: ${{ secrets.GITHUB_TOKEN }}
+         quayUser: ${{ secrets.QUAY_USERNAME }}
+         quayPass: ${{ secrets.QUAY_TOKEN }}
+      with:
+         repository: ${{ github.repository }}
+         runId: ${{ github.event.workflow_run.id }}
+         serverArtifact: infinispan-dist
+         tags:  ${{ needs.get-server-version.outputs.imageName }}:gh-${{ needs.get-info.outputs.sourceHeadBranch }}
```

Comment:
AFAICT the only difference between these two jobs is the `tags` field. We can calculate the tags as part of the `get-server-version` job, or a dedicated job and then use the output so that we only need a single job fro both `push` and `pull-request` events.